src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -264,11 +265,33 @@ public void resetStartsAndStops() {
         assertNotNull(p.getProperty(DisableConcurrentBuildsJobProperty.class));
         assertNull(p.getProperty(BuildDiscarderProperty.class));
 
-        JobPropertyTrackerAction action3 = b3.getAction(JobPropertyTrackerAction.class);
+        JobPropertyTrackerAction action3 = p.getAction(JobPropertyTrackerAction.class);
         assertNotNull(action3);
         assertTrue(action3.getJobPropertyDescriptors().isEmpty());
     }
 
+    @Issue("JENKINS-44848")
+    @LocalData
+    @Test
+    public void trackerPropertyUpgrade() throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("trackerPropertyUpgrade", WorkflowJob.class);
+        assertNotNull(p);
+        WorkflowRun b1 = p.getLastBuild();
+        assertNotNull(b1);
+        assertNotNull(p.getProperty(BuildDiscarderProperty.class));
+        assertNull(p.getAction(JobPropertyTrackerAction.class));
+
+        p.setDefinition(new CpsFlowDefinition("properties([disableConcurrentBuilds()])", true));
+
+        r.buildAndAssertSuccess(p);
+
+        assertNull(p.getProperty(BuildDiscarderProperty.class));
```

Comment:
Correct - `properties` was called in `b1`.

I think I can tweak `onlyRemoveJenkinsfileProperties` to cover the cases you mention - if there's no `JobPropertyTrackerAction`, then there's no difference between a build without the `properties` step having been run and/or no completed builds.